connect/mirror/src/main/java/org/apache/kafka/connect/mirror/OffsetSyncStore.java

```diff
@@ -105,7 +106,13 @@ private KafkaBasedLog<byte[], byte[]> createBackingStore(MirrorCheckpointConfig
     /**
      * Start the OffsetSyncStore, blocking until all previous Offset Syncs have been read from backing storage.
      */
-    public void start() {
+    public void start(boolean initializationMustReadToEnd) {
+        this.initializationMustReadToEnd = initializationMustReadToEnd;
+        if (initializationMustReadToEnd) {
+            log.warn("OffsetSyncStore initializationMustReadToEnd = {}", initializationMustReadToEnd);
```

Comment:
debug level, this is not worth warning about.

:+1: for the variable name

Reply:
Our intention was to alert the Kafka admin that they are getting degraded behavior. However, reflecting on this, I wonder if would be better to special case "not authorized" such that:
1. For not authorized we emit a warning (with a better worded explanation) - because the Kafka admin can choose to take an action that improves the frequency of checkpoints.
1. For other cases, log at debug level.

Reply:
I think that getting the fallback degraded behavior is not the normal case and therefore I strongly prefer a warn here.
However here we do know the reason why we are getting this - so we should explain that in the record handler as suggested